### PR TITLE
phpdoc fixes

### PR DESCRIPTION
**Pull request by DaSourcerer**
This PR fixes and enhances the existing phpdoc. Still far from really being phpdoc-complete, but it's enough to let any autocomplete implementation work sufficiently.

### CODE REVIEW

**Comment by jbowens** [comment]
lgtm, thanks

**Comment by Kubo2** [comment]
@DaSourcerer :+1:

File: JBBCode/CodeDefinition.php [hunk 1]
```diff
@@ -11,28 +11,28 @@
  */
 class CodeDefinition
 {
-    /* NOTE: THIS PROPERTY SHOULD ALWAYS BE LOWERCASE; USE setTagName() TO ENSURE THIS */
+    /* @var string NOTE: THIS PROPERTY SHOULD ALWAYS BE LOWERCASE; USE setTagName() TO ENSURE THIS */
```
Comment (Art4) [inline JBBCode/CodeDefinition.php]: `/*` must be `/**` imho.

Comment (Art4) [inline JBBCode/CodeDefinition.php]: Same in line 17, 20, 23, and so on.

Comment (DaSourcerer) [inline JBBCode/CodeDefinition.php]: Nope, that's the inline style recognized by most (all?) IDEs. If you're experiencing any troubles, I'll gladly "fix" it, though.

Comment (Art4) [inline JBBCode/CodeDefinition.php]: Maybe IDEs will recognize it, but phpdoc needs `/**`, see http://phpdoc.org/docs/latest/references/phpdoc/tags/var.html#examples

Comment (DaSourcerer) [inline JBBCode/CodeDefinition.php]: Huh, I thought this changed in PSR 5; it appears I was wrong. Going to change this as soon as I get home.

Comment (DaSourcerer) [inline JBBCode/CodeDefinition.php]: For reference: https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#55-examples

**Pull request merged**